zebrad/src/components/inbound.rs

```diff
@@ -178,6 +192,14 @@ impl Service<zn::Request> for Inbound {
                 None => async { Err("not ready to serve addresses".into()) }.boxed(),
```

Comment:
Why do we error here, but return Response::Nil when downloads is missing?

Reply:
We want to consistently return Response::Nil, because the peer logs an error when we return an error:
https://github.com/ZcashFoundation/zebra/blob/main/zebra-network/src/peer/connection.rs#L888
but we just want an info-level message saying that we dropped an early peer request.

We can add these info-level messages to the address book and downloads convenience methods.